### PR TITLE
py-tensorflow: change py-tensorflow@2.16-rocm-enhanced to use tarball instead of branch

### PR DESCRIPTION
**Pull request by afzpatel**
A rocm enhanced tf@2.16 tarball is now available so we'll switch to using that instead of the branch
https://github.com/ROCm/tensorflow-upstream/releases/tag/r2.16-rocm-enhanced

Also the Add_rocm_libs_path.patch file can be removed and replaced with this commit: https://github.com/ROCm/tensorflow-upstream/commit/8b7fcccb2914078737689347540cb79ace579bbb

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-tensorflow/package.py [hunk 1]
```diff
@@ -418,6 +418,8 @@ class PyTensorflow(Package, CudaPackage, ROCmPackage, PythonExtension):
     # and https://github.com/abseil/abseil-cpp/issues/1665
     patch("absl_neon.patch", when="@2.16.1: target=aarch64:")
 
+    # reverting change otherwise the commit patches won't apply
+    patch("revert_fd6b0a4.patch", when="@2.16-rocm-enhanced +rocm")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-tensorflow/package.py]: Tell me more about this one. Is this commit not needed? Which other patches don't apply? Can we change the order in which the patches are applied? Also see #43040 if it's helpful.

Comment (afzpatel) [reply var/spack/repos/builtin/packages/py-tensorflow/package.py]: The c467913 commit patch won't apply without first reverting fd6b0a4. 
Also thanks for the link, I've replaced the patch file now and I'm using the github commit url patch instead.

**Review by adamjstewart — APPROVED** [review]
Maybe a good next step would be to get `py-tensorflow+rocm` building in CI?

**Comment by afzpatel** [comment]
> Maybe a good next step would be to get `py-tensorflow+rocm` building in CI?

Yes, that sounds good.

**Comment by adamjstewart** [comment]
You basically just need to revert https://github.com/spack/spack/pull/43825 and get CI to pass.

**Pull request merged**